.github/workflows/ci.yml

```diff
@@ -19,62 +19,118 @@ jobs:
         include:
           - operating-system: 'ubuntu-20.04'
             php-version: '7.4'
-            job-description: 'with lowest deps'
+            job-description: 'tests with Symfony ^5'
+            run-tests: 'yes'
+            execute-flex-with-symfony-version: '^5' # explicit check for Symfony 5.x compatibility
+
+          - operating-system: 'ubuntu-20.04'
+            php-version: '7.4'
+            job-description: 'Fixer with lowest deps'
+            fixer-mode: 'standard'
             composer-flags: '--prefer-stable --prefer-lowest' # should be checked on the lowest supported PHP version
 
           - operating-system: 'ubuntu-20.04'
             php-version: '7.4'
-            job-description: 'with PHPDoc to type rules'
-            phpdoc-to-type-rules: 'yes' # should be checked on the lowest supported PHP version
+            job-description: 'Fixer with PHPDoc to type rules'
+            fixer-mode: 'phpdoc-to-type-rules' # should be checked on the lowest supported PHP version
 
           - operating-system: 'ubuntu-20.04'
             php-version: '7.4'
-            job-description: 'with Symfony ^5'
+            job-description: 'Fixer with Symfony ^5'
+            fixer-mode: 'standard'
             execute-flex-with-symfony-version: '^5' # explicit check for Symfony 5.x compatibility
 
           - operating-system: 'ubuntu-20.04'
             php-version: '8.0'
+            job-description: 'Fixer'
+            fixer-mode: 'standard'
+
+          - operating-system: 'ubuntu-20.04'
+            php-version: '8.0'
+            job-description: 'tests'
+            run-tests: 'yes'
 
           - operating-system: 'ubuntu-20.04'
             php-version: '8.1'
-            job-description: 'with Symfony ^6'
+            job-description: 'Fixer with Symfony ^6'
+            fixer-mode: 'standard'
+            execute-flex-with-symfony-version: '^6' # explicit check for Symfony 6.x compatibility
+
+          - operating-system: 'ubuntu-20.04'
+            php-version: '8.1'
+            job-description: 'tests with Symfony ^6'
+            run-tests: 'yes'
             execute-flex-with-symfony-version: '^6' # explicit check for Symfony 6.x compatibility
 
+          - operating-system: 'ubuntu-20.04'
+            php-version: '8.2'
+            job-description: 'Fixer'
+            fixer-mode: 'standard'
+
           - operating-system: 'ubuntu-20.04'
             php-version: '8.2'
             job-description: 'with migration rules'
-            migration-rules: 'yes' # should be checked on the highest supported PHP version
+            fixer-mode: 'migration-rules' # should be checked on the highest supported PHP version
```

Comment:
very similar concern as https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7366/files#r1370886849

originally, we were checking how app will behave after applying those rules - we run PHPUnit and Fixer itself.
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/actions/runs/6605602880/job/17940938431

now, we apply migration rules, but we do not test how app behaves afterwards
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/actions/runs/6619129235/job/17979005364


those rules are risky and breaking things, potentially.

Reply:
OK, this one is my bad, I did not realise `Run PHP CS Fixer with migration rules` actually applies changes. Will fix it when we confirm [this](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7366#discussion_r1371977133).

Reply:
in contrary to LOWEST job (https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7366#discussion_r1372073524),
HIGHEST job shall apply changes and see if all would work with those changes - but not expect those changes to be already applied.

Reply:
I've improved conditions for job matrix and now Fixer is running after applying migration rules on highest supported PHP version: 

<img width="1728" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/c5d439c0-d21f-46e6-812c-2e46f73d77e9">